CONTRIBUTING.md

```diff
@@ -0,0 +1,205 @@
+If your issue doesn’t already exist, and you’re ready to create a new one, make sure to state what you would like to implement, improve or bugfix. We have provided a template to make this process easier for you.
+
+**To open a Github issue, go to the RasaHQ repository, select “Issues”, “New Issue” then “Feature Request” or “Bug Report” and fill out the template.**
+
+![](https://www.rasa.com/assets/img/contributor-guidelines/opening-new-issue.png)
+
+The Rasa team will then get in touch with you to discuss if the proposed feature aligns with the company's roadmap, and we will guide you along the way in shaping the proposed feature so that it could be merged to the Rasa codebase.
+
+### Working on an existing feature
+
+If you choose to work on an existing issue, check out the Rasa contributors board to find existing open issues you can work on.
+
+The issues are handpicked by the Rasa team to have labels which correspond to the difficulty/estimated time needed to resolve the issue.
+
+**To work on an existing issue, go to the contributor project board, add a comment stating you would like to work on it and include any solutions you may already have in mind.**
+
+![](https://www.rasa.com/assets/img/contributor-guidelines/exiting-issue-sara.png)
+
+Someone from Rasa will then assign that issue to you and help you along the way.
+
+---
+
+## How to open a GitHub Pull Request
+
+### What is a Pull Request (PR)?
+
+This is how the GitHub team defines a PR:
+
+> “Pull requests let you tell others about changes you’ve pushed to a branch in a repository on GitHub. Once a pull request is opened, you can discuss and review the potential changes with collaborators and add follow-up commits before your changes are merged into the base branch.”
+
+This process is used by both Rasa team members and Rasa contributors to make changes and improvements to Rasa Open Source.
+
+### What to know before opening a PR
+
+#### Opening issues before PRs
+
+We usually recommend opening an issue before a pull request if there isn’t already an issue for the problem you’d like to solve. This helps facilitate a discussion before deciding on an implementation. See How to open a GitHub issue & file a bug report.
+
+#### Draft PRs
+
+To get some quick initial feedback from the Rasa team, you can create a draft pull request.
+
+#### PRs should be a reasonable length
+
+If your PR is greater than 500 lines, please consider splitting it into multiple smaller contributions.
+
+#### Code style
+
+To ensure a standardized code style we recommend using formatter black. To ensure our type annotations are correct we also suggest using the type checker pytype.
+Travis is a hosted continuous integration service used to build and test software projects hosted at GitHub. If your code is not formatted properly or doesn't type check, Travis will fail to build.
```

Comment:
Not using Travis anymore but rather github actions. Travis wasn't really introduced here as the thing that runs the tests 😀

I'd recommend putting this, formatting, and type checking in the same section, where i'd quick explain the tests that our CI/CD runs, and how to reproduce them locally. then you can just start the section with `make install-full` (the new `pip install -r requirements-dev.txt.`), and then mention how to run the tests as well as linting and formatting

Reply:
@erohmensing Great that you spotted this!

 Do we have some sort of documentation on them? Or do you have any advice on where to collect the information with description of the tests?

Reply:
You can see all the steps here: https://github.com/RasaHQ/rasa/blob/master/.github/workflows/continous-integration.yml (search for `run:`)

`make install-full` will install everything, then they can run the tests locally with e.g. 
`make lint` `make types` `make test`